final_project/UITestFixture.cs

```diff
@@ -17,21 +18,65 @@ public async Task Setup()
                 Headless = false
             });
 
+            string subPath = "../../../playwright/auth";
+            string filePath = "../../../playwright/auth/state.json";
+
+            if(!Directory.Exists(subPath))
+            {
+                Directory.CreateDirectory(subPath);
+            }
+
+            if(!File.Exists(filePath))
+            {
+                File.AppendAllText(filePath, "{}");
+            }
+
             context = await browser.NewContextAsync(new BrowserNewContextOptions
             {
                 ViewportSize = new ViewportSize
                 {
                     Width = 1920,
                     Height = 1080
-                }
+                },
+                StorageStatePath = "../../../playwright/auth/state.json"
+            });
+
+            await context.Tracing.StartAsync(new()
+            {
+                Title = $"{TestContext.CurrentContext.Test.ClassName}.{TestContext.CurrentContext.Test.Name}",
+                Screenshots = true,
+                Snapshots = true,
+                Sources = true
             });
 
             page = await context.NewPageAsync();
+            var homePage = new HomePage(page);
+            await homePage.Open();
+            if(await homePage.GetLoginPageLocator().IsVisibleAsync()) 
+            {
+                await homePage.AcceptDataPolicy();
+                await homePage.Login("bilikmike@gmail.com", "@vvwrER6FT5PgU9");
```

Comment:
here you need to use email and password as variables (the same values should be used in OneTimeSetup

Reply:
Done